### PR TITLE
add client token support for tracim web request

### PR DESCRIPTION
**Pull request by inkhey**
related to #3129 

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: backend/tracim_backend/lib/utils/request.py [hunk 1]
```diff
@@ -255,6 +260,11 @@ def set_user(self, user: User):
         super().set_user(user)
         self.plugin_manager.hook.on_current_user_set(user=user, request=self)
 
+    def set_client_id(self, client_id: str):
+        """Overload TracimContext method to add plugin hook call."""
+        super().set_client_id(client_id)
+        self.plugin_manager.hook.on_current_client_id_set(client_id=client_id, request=self)
+
```
Comment (grignards) [inline backend/tracim_backend/lib/utils/request.py:267]: I think we can define this method only on `TracimRequest` as it has a meaning only for HTTP requests coming from Tracim's frontend.

**Comment by inkhey** [comment]
@Skylsmoi Can you test this branch, set "X-Tracim-ClientId" Header and check TLM received ?

**Comment by lebouquetin** [comment]
I suggest:

- `_id` suffix -> should be used for int values
- `_token` or`_slug` suffix for str values.

It make things more clear.

**Comment by inkhey** [comment]
X-Tracim-ClientTo

> I suggest:
> 
>     * `_id` suffix -> should be used for int values
> 
>     * `_token` or`_slug` suffix for str values.
> 
> 
> It make things more clear.
> 
> edit: in the current case, this is not a slug (which implies a "static" value), so i suggest `_token`

So :
```
Header: X-Tracim-ClientToken
tlm field name: client_token
```

**Comment by lebouquetin** [comment]
@Skylsmoi yes, this is my suggestion. @inkhey @grignards do you agree with this?

**Comment by inkhey** [comment]
> @Skylsmoi yes, this is my suggestion. @inkhey @grignards do you agree with this?

I found "token" a bit less clear about the usage than id (a token is an id, but when i do think token, i mostly think about authentication), but that's not a big deal.

The reason i do use str here, is that i do think it's easier for frontend to create unique id here, maybe should we use uuid ? @Skylsmoi Did you suggest something specific for identificator here  ?

**Comment by lebouquetin** [comment]
> a token is an id, but when i do think token, i mostly think about authentication

I disagree that token is related to authentication; it is related to session for sure.

Using a string is required. You do not need to know in backend how it is used, actually. This is a feature usefull for clients but it has nothing to do with backend (it only allows to link API request with associated TLM)

**Comment by grignards** [comment]
> @Skylsmoi yes, this is my suggestion. @inkhey @grignards do you agree with this?

`Token` is ok for me.

**Comment by Skylsmoi** [comment]
I don't think I suggested anything

Since it's a header, it will be a string though. I don't plan to cast it to int, string is fine

**Comment by Skylsmoi** [comment]
@inkhey This is a full backend PR, what should I do exactly ?

**Comment by inkhey** [comment]
quality test here does no make sense : i checked.

**Pull request merged**